### ISSUE TITLE
how to generate comb uuid?

### ISSUE THREAD

**Issue by Daskul** [ramsey/uuid#53]
I've tried to generate atleast thousands of uuid using Uuid::uuid4() but they are not in sequence.

**Comment by aztech-dev** [comment]
Current release does not provide COMB support. You'll have to wait for the 3.0 release to generate COMB's.

**Comment by Daskul** [comment]
I see. I thought I've read before that it was already supported. Thank you for the information

**Comment by hi2u** [comment]
Now that it's in the library, it would be good if it was documented in the README.md 

I couldn't figure it out until I came across https://benramsey.com/blog/2016/04/ramsey-uuid/

```php
$factory = new \Ramsey\Uuid\UuidFactory();
$generator = new \Ramsey\Uuid\Generator\CombGenerator($factory->getRandomGenerator(), $factory->getNumberConverter());
$codec = new \Ramsey\Uuid\Codec\TimestampFirstCombCodec($factory->getUuidBuilder());
 
$factory->setRandomGenerator($generator);
$factory->setCodec($codec);
 
\Ramsey\Uuid\Uuid::setFactory($factory);
 
$uuid = \Ramsey\Uuid\Uuid::uuid4();
```

Is this the easiest way to do it?  

Would it also be worth adding a simple method to the main class such as Uuid::uuid4comb() that does it for us without pasting all the above into our own code?

**Comment by ramsey** [comment]
Thanks for the recommendation. I'm currently working on documentation for this and hope to have it posted soon.

**Comment by 1ma** [comment]
How feasible would it be to add a new static method to `Uuid` to generate this kind of UUIDs?

**Comment by ramsey** [comment]
@1ma That sounds like it could be reasonable to me. Something like `Uuid::comb()`?

**Comment by ramsey** [comment]
Sounds great. Thanks!

**Comment by kael-shipman** [comment]
Just a reminder: Looks like this never made it into the docs. I'm using @hi2u's implementation for now but would love to know if there's a "standard" way to do it.

**Comment by broofa** [comment]
FWIW, we (node-uuid) decided to pass on supporting "comb" uuids.  See https://github.com/kelektiv/node-uuid/issues/303#issuecomment-458757302

tl;dr: RFC4122 Version 1 UUIDs serve basically the same purpose.

**Comment by hi2u** [comment]
@broofa The difference is that UUIDv1 isn't actually in timestamp order. 

Generate a few and you'll see the characters at the start moving faster than the ones in the middle.  i.e. It's kind of like `ss:mm:hh dd-mm-yyyy` rather than `yyyy-mm-dd hh:mm:ss`.  That's why comb UUIDs came about.

**Comment by ramsey** [comment]
V1 UUIDs should not appear to be ascending. They would get scattered in your DB index because of the way the time fields are reordered. I’m not sure why that benchmark shows them to be almost identical to COMB sequential v1 UUIDs in performance. Perhaps it’s because they do sort of appear sequential when generated quickly over a short period of time.

**Comment by broofa** [comment]
Hmm... 'Not sure I follow.  UUID fields start with `time_lo`, `time_mid`, then `time_high`... i.e low-order time fields correspond to low-order bytes, no?
All other fields being equal (i.e.. `node` and `clock_seq`)
Barring endian-ness of all this, doesn't that suggest that v1 ids should be monotonically increasing?

**Comment by ramsey** [comment]
When you generate the time for the UUID, it's in this format:

```
time_high + time_mid + time_lo
```

Then, according to RFC 4122, we rearrange the bytes in the time integer to:

```
time_lo + time_mid + time_hi
```

So, the fields aren't in an order that can be guaranteed to increase monotonically.

> Not actually sure I want to care about it now, either.

I know how you feel. 😆

**Comment by ramsey** [comment]
Another discussion pointed me back to this, and I wanted to clarify something...

> Octet #0 is the most significant?

Yes. This is because bytes in UUIDs are arranged in "network byte order" (a.k.a. big endian).
